IDinsight/aaq-core
# Enable HTTPS for endpoints

Reviewer: @amiraliemami 

Estimate: 2 hrs

----

# Enable HTTPS for endpoints

## Ticket

Fixes: [AAQ-142](https://idinsight.atlassian.net/browse/AAQ-142)

## Description, Motivation and Context

### Goal

To enable HTTPS/TLS encryption for admin app and backend api

### Changes

- Added nginx reverse proxy. Backend is available as `/api` through reverse proxy
- Add certbot to generate certificates from letsencrypt
- docker-compose no longer exposes ports on the backend and admin app
- See new deployment instructions

## How Has This Been Tested?

Test cases pass. 
I have tested our dev setup to make sure that still works
I have run it using docker-compose to check that it works. 

Testing cert issuing using ngrok
1. Install and configure ngrok
2. Run `ngrok http https://localhost` to forward traffic
3. In `.env` file, ensure you have
```
NEXT_PUBLIC_BACKEND_URL=https://[NGROK URL]/api
BACKEND_ROOT_PATH="/api"
```
4. In `.env.nginx` file ensure you have
```
DOMAIN=[NGROK URL]  # don't add https/http at the front
```
6. Run `./init-letsencrypt.sh`
7. Run `docker compose`
```
docker compose -f docker-compose.yml -p aaq-stack up -d --build
```
8. Go to `https://[NGROK URL]` and make sure it all works

## To-do before merge

- [x] DONE - Ensure PR #32 is merged

## Checklist:

This checklist is a useful reminder of small things that can easily be forgotten.
Put an `x` in all the items that apply and remove any items that are not relevant to this PR.


[1]: http://chris.beams.io/posts/git-commit/
